doc/source/tune/tutorials/tune-trial-checkpoint.rst

```diff
@@ -0,0 +1,132 @@
+
+**Manual Checkpointing**: A custom Trainable can manually trigger checkpointing by returning ``should_checkpoint: True``
+(or ``tune.result.SHOULD_CHECKPOINT: True``) in the result dictionary of `step`.
+This can be especially helpful in spot instances:
+
+.. code-block:: python
+
+    def step(self):
+        # training code
+        result = {"mean_accuracy": accuracy}
+        if detect_instance_preemption():
+            result.update(should_checkpoint=True)
+        return result
+
+
+**Periodic Checkpointing**: periodic checkpointing can be used to provide fault-tolerance for experiments.
+This can be enabled by setting ``checkpoint_frequency=<int>`` and ``max_failures=<int>`` to checkpoint trials
+every *N* iterations and recover from up to *M* crashes per trial, e.g.:
+
+.. code-block:: python
+
+    tuner = tune.Tuner(
+        my_trainable,
+        run_config=air.RunConfig(
+            checkpoint_config=air.CheckpointConfig(checkpoint_frequency=10),
+            failure_config=air.FailureConfig(max_failures=5))
+    )
+    results = tuner.fit()
+
+**Checkpointing at Termination**: The checkpoint_frequency may not coincide with the exact end of an experiment.
+If you want a checkpoint to be created at the end of a trial, you can additionally set the ``checkpoint_at_end=True``:
+
+.. code-block:: python
+   :emphasize-lines: 5
+
+    tuner = tune.Tuner(
+        my_trainable,
+        run_config=air.RunConfig(
+            checkpoint_config=air.CheckpointConfig(checkpoint_frequency=10, checkpoint_at_end=True),
+            failure_config=air.FailureConfig(max_failures=5))
+    )
+    results = tuner.fit()
+
+
+Use ``validate_save_restore`` to catch ``save_checkpoint``/``load_checkpoint`` errors before execution.
+
+.. code-block:: python
+
+    from ray.tune.utils import validate_save_restore
+
+    # both of these should return
+    validate_save_restore(MyTrainableClass)
+    validate_save_restore(MyTrainableClass, use_object_store=True)
+
+Configurations
+==============
+Checkpointing can be configured through :class:`CheckpointConfig <ray.air.config.CheckpointConfig>`.
+Some of the options may not apply to Function Trainable. See table below.
+
+.. list-table::
+   :header-rows: 1
+
+   * -
+     - Function API
+     - Class API
+   * - ``num_to_keep``
+     - |:heavy_check_mark:|
+     - |:heavy_check_mark:|
+   * - ``checkpoint_score_attribute``
+     - |:heavy_check_mark:|
+     - |:heavy_check_mark:|
+   * - ``checkpoint_score_order``
+     - |:heavy_check_mark:|
+     - |:heavy_check_mark:|
+   * - ``checkpoint_frequency``
+     - |:heavy_multiplication_x:|
+     - |:heavy_check_mark:|
+   * - ``checkpoint_at_end``
+     - |:heavy_multiplication_x:|
+     - |:heavy_check_mark:|
+
+
+
+See also
+========
+:ref:`Storage configuration guide <tune-storage-options>`
+:ref:`Running Distributed Experiments with Ray Tune <tune-distributed-ref>`
```

Comment:
```suggestion
Summary
=======

In this user guide, we covered how to save and load trial checkpoints in Tune.
Once checkpointing has been enabled, move onto one of the following guides to find out how to:

- :ref:`Extract checkpoints from Tune experiment results <>` --> link to https://docs.ray.io/en/latest/tune/examples/tune_analyze_results.html
- :ref:`Configure persistent storage options` for a :ref:`distributed Tune experiment <tune-distributed-ref>`
- Perform batch prediction or serving --> link to AIR user guides
```

Reply:
decided that batch prediction/serving not that related to this added. but added the other two.